scripts/Muon/GUI/Common/muon_group_pair_context.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import (absolute_import, division, print_function)
+
+import Muon.GUI.Common.utilities.xml_utils as xml_utils
+from Muon.GUI.Common.muon_group import MuonGroup
+from Muon.GUI.Common.muon_pair import MuonPair
+
+from mantid.api import WorkspaceGroup
+from mantid.kernel import ConfigServiceImpl
+from Muon.GUI.Common.observer_pattern import Observable
+
+
+def get_default_grouping(workspace, instrument, main_field_direction):
+    parameter_name = "Default grouping file"
+    if instrument == "MUSR" or instrument == 'CHRONUS':
+        parameter_name += " - " + main_field_direction
+    try:
+        if isinstance(workspace, WorkspaceGroup):
+            grouping_file = workspace[0].getInstrument().getStringParameter(parameter_name)[0]
+        else:
+            grouping_file = workspace.getInstrument().getStringParameter(parameter_name)[0]
+    except IndexError:
+        return [], []
+    instrument_directory = ConfigServiceImpl.Instance().getInstrumentDirectory()
+    filename = instrument_directory + grouping_file
+    new_groups, new_pairs, description = xml_utils.load_grouping_from_XML(filename)
+    return new_groups, new_pairs
+
+
+def construct_empty_group(group_names, group_index=0):
```

Comment:
do we need this since we force the user to name the group?

Reply:
This is not currently used as we do not allow users to rename groups/pairs. I have left it in for now though because with the changes this is now possible to do if we want to re-enable that column in the GUI